src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets

```diff
@@ -41,6 +42,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ImportNuGetBuildTasksPackTargetsFromSdk Condition="'$(ImportNuGetBuildTasksPackTargetsFromSdk)' == ''">false</ImportNuGetBuildTasksPackTargetsFromSdk>
     <AllowedOutputExtensionsInPackageBuildOutputFolder>.dll; .exe; .winmd; .json; .pri; .xml; $(AllowedOutputExtensionsInPackageBuildOutputFolder)</AllowedOutputExtensionsInPackageBuildOutputFolder>
     <AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder>.pdb; .mdb; $(AllowedOutputExtensionsInPackageBuildOutputFolder); $(AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder)</AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder>
+    <ResolveProjectReferenceVersionDuringPack Condition="'$(ResolveProjectReferenceVersionDuringPack)' == ''">false</ResolveProjectReferenceVersionDuringPack>
```

Comment:
`false` should not be the default. I can't imagine any case where this would produce a correct result. For p2p versions, I think the property is the only guaranteed true/correct value.

Reply:
@onovotny any package that aims to change project versions during build time can set this property in a props file and turn on this behavior. 
Regardless of which is the more correct behavior, i feel this is still is a breaking change and hence the opt-in flag.

CC: @rrelyea

Reply:
Is it a breaking change or fixing behavior that has always been broken?

Reply:
I would still classify that as a breaking change- but let me run it with @rrelyea and get back.